file2.js

```diff
@@ -0,0 +1,558 @@
+
+          return langExtensions(languageName.toLowerCase(), extensions);
+        });
+
+        var postData = {};
+        postData.projectId = projectId;
+        postData.languageExtensions = JSON.stringify(arr);
+
+        utils.post(url, postData, function(e) {
+          if (e.error) {
+            $(".jsLanguageExtensionFeedback").showError(e.error);
+          } else {
+            $(".jsLanguageExtensionFeedback").showSuccess("Saved");
+            ProjectSettings.prototype.changesFooter("#changesFooter-extensions");
+          }
+        }, function() {
+          $(".jsLanguageExtensionFeedback").showError("Could not save your settings please try again.");
+        });
+      });
+
+    };
+
+    ProjectSettings.prototype.bindGenerateKey = function() {
+      $(".btnGenerateKey").click(function() {
+        $(this).addClass('disabled');
+        var url = $(this).data("url");
+        utils.post(url, null, function(e) {
+          if (e.error) {
+            $(".GenerateKeyFeedback").showError(e.error);
+          } else {
+            location.reload();
+          }
+        }, function() {
+          $(".GenerateKeyFeedback").showError("Could not generate key, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindGenerateUserKey = function() {
+      $(".btnGenerateUserKey").click(function() {
+        $(this).addClass('disabled');
+        var url = $(this).data("url");
+        utils.post(url, null, function(e) {
+          if (e.error) {
+            $(".GenerateUserKeyFeedback").showError(e.error);
+          } else {
+            location.reload();
+          }
+        }, function() {
+          $(".GenerateUserKeyFeedback").showError("Could not add key to user, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindRemoveProject = function() {
+      $(".btnRemoveProject").click(function() {
+        var $this = $(this);
+        var removeUrl = $this.data("url");
+        var redirectUrl = $this.data("iden");
+
+
+        utils.post(removeUrl, {}, function(e) {
+          if (e.error) {
+            $(".RemoveProjectFeedback").showError(e.error);
+          } else {
+            window.location.href = redirectUrl;
+          }
+        }, function() {
+          $(".RemoveProjectFeedback").showError("Could not remove project, please try again.");
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindTransferProject = function() {
+
+      var $orgBox = $("select.js-project-destination");
+      var $teamBox = $("select.js-organization-teams-selection-box");
+
+      $(".btnTransferProject").click(function() {
+        var $this = $(this);
+
+        var payload = {};
+
+        payload.orgId = $orgBox.val();
+
+        var teamIds = $teamBox.val();
+
+        if (teamIds) {
+          payload.teamIds = teamIds;
+          var url = $this.data("url");
+          utils.JSend(url, payload, function() {
```

Comment:
![Codacy](https://app.dev.codacy.org/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)](https://app.dev.codacy.org/app/codacytestuser/CheckPrCommentsPublicProject/pullRequest?prid=8021)